acceptance/lib/puppet/acceptance/mount_utils.rb

```diff
@@ -0,0 +1,56 @@
+module Puppet
+  module Acceptance
+    module MountUtils
+
+      # Return the absolute path to the filesystem table file.
+      # @param host [String] hostname
+      # @return [String] path to the filesystem table file.
+      def filesystem_file(host)
+        if host['platform'] =~ /aix/
+          '/etc/filesystems'
+        else
+          '/etc/fstab'
+        end
+      end
+
+      # Return a standard filesystem type to use when creating filesysytems.
+      # @param host [String] hostname
+      # @return [String] filesystem type.
+      def filesystem_type(host)
+        if host['platform'] =~ /aix/
+          'jfs2'
+        else
+          'ext3'
+        end
+      end
+
+      # Appends a new filesystem entry to the filesystem table.
+      # @param host [String] hostname.
+      # @param mount_name [String] the name of the mount point. We use /tmp/name as the
+      # new filesystem, and /name as the actual mount point.
+      def add_entry_to_filesystem_table(host, mount_name)
+        fs_file = filesystem_file(host)
+
+        if host['platform'] =~ /aix/
+          on(host, "echo '/#{mount_name}:\n  dev = /tmp/#{mount_name}\n  vfs = jfs2' >> #{fs_file}")
+        else
+          on(host, "echo '/tmp/#{mount_name}  /#{mount_name}  ext3  loop  0  0' >> #{fs_file}")
+        end
+      end
+
+      # Creates a new filesystem on the host.
+      # @param host [String] hostname
+      # @param mount_name [String] the name of the mount point.
+      # Note: On AIX, `crfs` creates the filesystem *and* adds it to /etc/filesystems.
+      def create_filesystem(host, mount_name)
+        if host['platform'] =~ /aix/
+          volume_group = on(host, 'lsvg').stdout.split("\n")[0]
+          on(host, "crfs -v jfs2 -a size=1M -g #{volume_group} -m /#{mount_name}")
+        else
+          on(host, "dd if=/dev/zero of=/tmp/#{mount_name} count=10240", :acceptable_exit_codes => [0,1])
+          on(host, "yes | mkfs -t ext3 -q /tmp/#{mount_name}", :acceptable_exit_codes => (0..254))
```

Comment:
Should this also be more guarded now that it is in a helper? See comment for line 12.